cmd/check_container.go

```diff
@@ -157,106 +155,43 @@ var checkContainerCmd = &cobra.Command{
 			log.Info("preparing results that will be submitted to Red Hat")
 			log.Tracef("CertProject: %+v", certProject) // TODO() This depends on Pyxis calls made outside of this block.
 
-			// The engine writes the certified image config to disk in a Pyxis-specific format.
-			// Include that data in our submission.
-			certImageFilepath := path.Join(artifacts.Path(), certification.DefaultCertImageFilename)
-			log.Tracef("Loading certImage data from path %s", certImageFilepath)
-			certImageBytes, err := os.ReadFile(certImageFilepath)
+			submission, err := pyxis.NewCertificationInput(
+				certProject, // Note: certProject depends on Pyxis calls made outside of this block.
+				// The certification engine writes the rpmManifest for images not based on scratch.
+				// Include that data in our submission.
+				path.Join(artifacts.Path(), certification.DefaultRPMManifestFilename),
+				// The engine writes the certified image config to disk in a Pyxis-specific format.
+				// Include that data in our submission.
+				path.Join(artifacts.Path(), certification.DefaultCertImageFilename),
+				// Include Preflight's test results in our submission. pyxis.TestResults embeds them.
+				path.Join(artifacts.Path(), certification.DefaultTestResultsFilename),
+			)
 			if err != nil {
 				return fmt.Errorf(
-					"%w: unable to read file from disk to include in submission: %s: %s",
+					"%w: could not build submission with required assets: %s",
 					errors.ErrSubmittingToPyxis,
-					certImageFilepath,
 					err,
 				)
 			}
 
-			var certImage pyxis.CertImage
-			err = json.Unmarshal(certImageBytes, &certImage)
-			if err != nil {
-				return fmt.Errorf(
-					"%w: data for the %s appears to be malformed: %s",
-					errors.ErrSubmittingToPyxis,
-					"certImage",
-					err,
-				)
-			}
-
-			// The certification engine also writes the rpmManifest for images not based on scratch.
-			// Include that data in our submission.
-			rpmManifestFilepath := path.Join(artifacts.Path(), certification.DefaultRPMManifestFilename)
-			log.Tracef("Loading RPM manifest data from path %s", rpmManifestFilepath)
-			rpmManifestBytes, err := os.ReadFile(rpmManifestFilepath)
-			if err != nil {
-				return fmt.Errorf(
-					"%w: unable to read file from disk to include in submission: %s: %s",
-					errors.ErrSubmittingToPyxis,
-					rpmManifestFilepath,
-					err,
-				)
-			}
-
-			var rpmManifest pyxis.RPMManifest
-			err = json.Unmarshal(rpmManifestBytes, &rpmManifest)
-			if err != nil {
-				return fmt.Errorf(
-					"%w: data for the %s appears to be malformed: %s",
-					errors.ErrSubmittingToPyxis,
-					"rpm manifest",
-					err,
-				)
-			}
-
-			// Include Preflight's test results in our submission. pyxis.TestResults embeds them.
-			var testResults pyxis.TestResults
-			err = json.Unmarshal(formattedResults, &testResults)
-			if err != nil {
+			if _, err := submission.WithArtifactFromFile(viper.GetString("logfile")); err != nil {
```

Comment:
So, when I think builder, I think of chaining these calls...

NewCertificationInput(certProject).WithArtifactFromFile("my/path").WithRPMManifest("manifestpath")... Not sure if that's possible in this case in order to handle errors... Each of the "With" methods would return the "certInput" with it's piece added to it.